redash/settings/__init__.py

```diff
@@ -106,9 +185,6 @@ def all_settings():
 STATIC_ASSETS_PATH = fix_assets_path(os.environ.get("REDASH_STATIC_ASSETS_PATH", "../client/dist/"))
 
 JOB_EXPIRY_TIME = int(os.environ.get("REDASH_JOB_EXPIRY_TIME", 3600 * 12))
-COOKIE_SECRET = os.environ.get("REDASH_COOKIE_SECRET", "c292a0a3aa32397cdb050e233733900f")
-SESSION_COOKIE_SECURE = parse_boolean(os.environ.get("REDASH_SESSION_COOKIE_SECURE") or str(ENFORCE_HTTPS))
-SECRET_KEY = os.environ.get('REDASH_SECRET_KEY', COOKIE_SECRET)
```

Comment:
`SECRET_KEY` is supposed to be used only for encrypting data source options. `COOKIE_SECRET` is the `COOKIE_SECRET`. While by default we use the same value for both, the idea was to allow setting a different one, if the user wants to.

From checking your PR, I realize that Flask calls this `SECRET_KEY` and hence the confusion here. 🤦‍♂️ 

This needs to be reverted to previous behavior.

Reply:
@arikfr Yeah, I understand the confusion, `SECRET_KEY` is automatically set to `app.secret_key` when we update `app.config` with the settings values after all. Plus we are currently using the value of `REDASH_COOKIE_SECRET` as the value for `SECRET_KEY`.

Since app.secret_key "will be used for securely signing the session cookie and can be used for **any other security related needs by extensions or your application**" [docs](http://flask.pocoo.org/docs/1.0/config/#SECRET_KEY) and it's often used in the extension community in that sense as well, shouldn't we introduce a `REDASH_DATASOURCE_SECRET_KEY` just for the encryption of the datasource secrets?

So basically shouldn't we have three things to be able to handle:

REDASH_SECRET_KEY = random string
REDASH_DATASOURCE_SECRET_KEY = random string
REDASH_COOKIE_SECRET_KEY = REDASH_SECRET_KEY or own string

Alternatively we could simply use REDASH_SECRET_KEY for the secure cookies and not have a separate setting REDASH_COOKIE_SECRET_KEY.

Either way, decoupling the encryption keys of the data source options from the session/cookie encryption keys seem like a good idea in case one of it is leaked and would reduce the impact.
Do you concur?

Reply:
@arikfr Quick ping, what I should do?

Reply:
> Yeah, I understand the confusion, SECRET_KEY is automatically set to app.secret_key when we update app.config with the settings values after all. 

But it was overridden back to `COOKIE_SECRET` value in `redash.authentication`.

While we do use the `COOKIE_SECRET` as the default value for the data source `SECRET_KEY`, it still should be decoupled to allow setting a different value.

So the updated code should be:

```python
SECRET_KEY = os.environ.get("REDASH_COOKIE_SECRET", "c292a0a3aa32397cdb050e233733900f")
DATASOURCE_SECRET_KEY = os.environ.get('REDASH_SECRET_KEY', SECRET_KEY)
```

And then update in `redash.models` to use `redash.settings.DATASOURCE_SECRET_KEY` instead of `redash.settings.SECRET_KEY`, while updating the Flask code to `redash.settings.SECRET_KEY`. There is no need to have `COOKIE_SECRET` anymore (and if we used it explicitly somewhere, just switch to SECRET_KEY).

This way everything stays backward compatible, while also consistent with Flask's naming.

Reply:
@arikfr Hm, from a user perspective I think that environment variable naming would be inconsistent with what they'd be used for.

With your proposal `REDASH_COOKIE_SECRET` is actually used as the Flask secret key, but `REDASH_SECRET_KEY` is used just for encrypting data source options. This seems like a guarantee for user confusion.

But we can easily prevent this by keeping `REDASH_SECRET_KEY` as the default setting for the one and only Flask secret key (and the one we promote in the docs) and treat `REDASH_COOKIE_SECRET` as deprecated.

```python
SECRET_KEY = os.environ.get(
    "REDASH_COOKIE_SECRET",
) or os.environ.get(
    "REDASH_SECRET_KEY",
    "c292a0a3aa32397cdb050e233733900f"
)  # using or here to cover for an empty REDASH_COOKIE_SECRET
```

And add a new `REDASH_DATASOURCE_SECRET_KEY` environment variable for the new feature to encrypt datasource options?

```python
DATASOURCE_SECRET_KEY = os.environ.get('REDASH_DATASOURCE_SECRET_KEY', SECRET_KEY)
```

Sorry for being a stickler here, I'm worried that we can't redo this later on :)

Reply:
I agree it's not ideal, and if I was aware of the setting in Flask before I would've picked a different name. But changing this now can break existing deployments, and therefore I don't see many options other than preserve it.

Reply:
The fallback above would still be backward compatible IIUC, but maybe I'm missing something. 🙈

Reply:
How will it support someone setting the `REDASH_SECRET_KEY` value to a different value than `REDASH_COOKIE_SECRET` to be used with data sources?

Reply:
In Redash v6 we only have `REDASH_COOKIE_SECRET` which they can keep using or switch to `REDASH_SECRET_KEY`.

And then optionally define `REDASH_DATASOURCE_SECRET_KEY` (which defaults to the value of the secret key above) when needed. For the setup scripts I suggest to also add `REDASH_DATASOURCE_SECRET_KEY` to `/opt/redash/env`.

So all those situations work:

1. Just using the Redash <= v6 name:
```
REDASH_COOKIE_SECRET="foo"
```

2. Just specifying the secret key with the new name:
```
REDASH_SECRET_KEY="foo"
```

3. Specifying both secret and encryptio key:
```
REDASH_SECRET_KEY="foo"
REDASH_DATASOURCE_SECRET_KEY="bar"
```

Reply:
> In Redash v6 we only have REDASH_COOKIE_SECRET

Yes, but there are people already using v7. I don't know if they used the new setting already or not, but prefer not to break their deployments.

Reply:
Aaaaah, that would break things, indeed. Sorry I was missing this detail :)

Reply:
Okay, updated this for the last time.